pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,491 @@
+/*
+Copyright 2024 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/duration"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/kubernetes/pkg/util/node"
+
+	"volcano.sh/apis/pkg/apis/batch/v1alpha1"
+	schedulingv1beta1 "volcano.sh/apis/pkg/apis/scheduling/v1beta1"
+	"volcano.sh/volcano/pkg/cli/util"
+)
+
+const (
+	// Name pod name
+	Name string = "Name"
+	// Ready pod ready
+	Ready string = "Ready"
+	// Status pod status
+	Status string = "Status"
+	// Restart pod restart
+	Restart string = "Restart"
+	// Age pod age
+	Age string = "Age"
+)
+
+type listFlags struct {
+	util.CommonFlags
+	// Namespace pod namespace
+	Namespace string
+	// JobName represents the pod created under this vcjob,
+	// filtered by volcano.sh/job-name label
+	// the default value is empty, which means
+	// that all pods under vcjob will be obtained.
+	JobName string
+	// allNamespace represents getting all namespaces
+	allNamespace bool
+	// QueueName represents queue name
+	QueueName string
+}
+
+var listPodFlags = &listFlags{}
+
+// InitListFlags init list command flags.
+func InitListFlags(cmd *cobra.Command) {
+	util.InitFlags(cmd, &listPodFlags.CommonFlags)
+
+	cmd.Flags().StringVarP(&listPodFlags.QueueName, "queue", "q", "", "list pod with specified queue name")
+	cmd.Flags().StringVarP(&listPodFlags.JobName, "job", "j", "", "list pod with specified job name")
+	cmd.Flags().StringVarP(&listPodFlags.Namespace, "namespace", "n", "default", "the namespace of job")
+	cmd.Flags().BoolVarP(&listPodFlags.allNamespace, "all-namespaces", "", false, "list jobs in all namespaces")
+}
+
+// ListPods lists all pods details created by vcjob
+func ListPods(ctx context.Context) error {
+	config, err := util.BuildConfig(listPodFlags.Master, listPodFlags.Kubeconfig)
+	if err != nil {
+		return err
+	}
+	if listPodFlags.allNamespace {
+		listPodFlags.Namespace = ""
+	}
+
+	var pods corev1.PodList
+	listPodFunc := func(labelSelector labels.Selector) (*corev1.PodList, error) {
```

Comment:
We can move this to a seperate func, the `ListPods` func is too long.

Reply:
done